x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -189,4 +192,29 @@ static DeprecationIssue checkTemplatesWithMultipleTypes(ClusterState state) {
             + " define multiple types and so will cause errors when used in index creation"
             );
     }
+
+    static DeprecationIssue checkFollowedSystemIndices(ClusterState state) {
+        Set<String> systemIndexFollowers = new HashSet<>();
+        for (ObjectObjectCursor<String, IndexMetadata> indexEntry : state.metadata().getIndices()) {
+            final IndexMetadata indexMetadata = indexEntry.value;
+            final Map<String, String> ccrMetadata = indexMetadata.getCustomData(CCR.CCR_CUSTOM_METADATA_KEY);
+            if (ccrMetadata == null) {
+                continue;
+            }
+            if (indexMetadata.isSystem()) {
+                systemIndexFollowers.add(indexEntry.key);
```

Comment:
Same comment on the scope as above, we should only log deprecation issues when the index was originally auto-followed I think?